src/Mjolnir.ts

```diff
@@ -399,13 +401,87 @@ export class Mjolnir {
         }
     }
 
+    private async loadSettings(): Promise<any> {
+        try {
+            this.settings = await this.client.getRoomStateEvent(
+                config.managementRoom, 'org.matrix.mjolnir.settings', ""
+            )
+        } catch { /* do nothing, we've got no settings */ }
+    }
+    private async saveSettings(): Promise<any> {
+        await this.client.sendStateEvent(
+            config.managementRoom, 'org.matrix.mjolnir.settings', "", this.settings
+        );
+    }
+
+    /*
+     * Read org.matrix.mjolnir.setting state object, find any saved settings for
+     * the requested protectionName, then iterate and validate against their parser
+     * counterparts in IProtection.settings and return those which validate
+     *
+     * @param protectionName The name of the protection whose settings we're reading
+     * @returns Every saved setting for this protectionName that has a valid value
+     */
+    public getProtectionSettings(protectionName: string): { [setting: string]: any } {
+        if (!(protectionName in this.settings)) return {};
```

Comment:
Ah, sorry for the lack of clarity, that's not what I meant in my previous review.

I meant to use
```js
this.client.getRoomStateEvent(config.managementRoom, /* state event name */ 'org.matrix.mjolnir.setting', /* state key */ protectionName)
```

This way, we still have one event/object per protection (which should keep them readable), but they're all `org.matrix.mjolnir.setting` (which should make them easier to inspect, backup, etc.).